application/src/main/java/org/opentripplanner/model/TripTimeOnDate.java

```diff
@@ -59,9 +64,15 @@ public TripTimeOnDate(
   /**
    * Must pass in both Timetable and Trip, because TripTimes do not have a reference to
    * StopPatterns.
+   *
+   * @return null if the trip does not exist in the timetable
    */
+  @Nullable
   public static List<TripTimeOnDate> fromTripTimes(Timetable table, Trip trip) {
     TripTimes times = table.getTripTimes(trip);
+    if (times == null) {
+      return null;
+    }
```

Comment:
How does this happen? Is there something wrong in the realtime updates?

Reply:
If the trip is an added trip, and we are checking its scheduled timetable, it will happen because an added trip doesn't have a scheduled timetable.